src/state/routing/slice.ts

```diff
@@ -43,22 +43,32 @@ export const routing = createApi({
             })
             const response = await global.fetch(`${args.routerUrl}quote?${query}`)
             if (!response.ok) {
-              return { error: { status: response.status, data: await response.text() } }
+              let data: string | Record<string, unknown> = await response.text()
+              try {
+                data = JSON.parse(data)
+              } catch {}
+
+              // NO_ROUTE should be treated as a valid response to prevent retries.
+              if (typeof data === 'object' && data.errorCode === 'NO_ROUTE') {
+                return { data: NO_ROUTE as GetQuoteResult }
+              }
+
+              throw data
             }
 
             const data: GetQuoteResult = await response.json()
             return { data }
```

Comment:
i don't love that this is just called "data"

Reply:
very fair - I've renamed it to `quote` in both cases where it is returned as valid data.
the name `data` is required by the typing of redux.